File: python/009.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import math

def p9(n):
    return list(map(
        lambda l: l[0] * l[1] * l[2],
        filter(
            lambda l: l[0] * l[0] + l[1] * l[1] == l[2] * l[2],
            [ (a, b, n - a - b)
              for a in range(1, int(n / 3)) for b in range(a, int(n / 2)) ])))

def main():
    print(p9(1000))

if __name__ == '__main__':
    main()

File: elixir/009.exs
#!/usr/bin/env elixir

# https://projecteuler.net/problem=9

defmodule P9 do
	def solve(n) do
		Enum.to_list(for a <- 1..div(n, 3), b <- a..div(n, 2), do: {a, b})
		|> Enum.filter_map(
			fn ({a, b}) ->
				c = (n - a - b)
				a * a + b * b == c * c
			end,
			fn ({a, b}) ->
				a * b * (n - a - b)
			end
		)
	end
end

P9.solve(1000)
|> IO.inspect
